examples/dialout-chatbot/bot_runner.py

```diff
@@ -0,0 +1,182 @@
+
+app = FastAPI(lifespan=lifespan)
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"]
+)
+
+"""
+Create Daily room, tell the bot if the room is created for Twilio's SIP or Daily's SIP (vendor).
+When the vendor is Daily, the bot handles the call forwarding automatically,
+i.e, forwards the call from the "hold music state" to the Daily Room's SIP URI.
+
+Alternatively, when the vendor is Twilio (not Daily), the bot is responsible for
+updating the state on Twilio. So when `dialin-ready` fires, it takes appropriate
+action using the Twilio Client library.
+"""
+
+
+async def _create_daily_room(room_url, callId, phone_number, callDomain=None):
+
+    if not room_url:
+        params = DailyRoomParams(
+            properties=DailyRoomProperties(
+                # Note: these are the default values, except for the display name
+                sip=DailyRoomSipParams(
+                    display_name="dialin-user",
+                    video=False,
+                    sip_mode="dial-in",
+                    num_endpoints=1,
+                ),
+                enable_dialout=True,
+            )
+        )
+
+        print("Creating new room...")
+        room: DailyRoomObject = await daily_helpers["rest"].create_room(params=params)
+
+    else:
+        # Check passed room URL exist (we assume that it already has a sip set up!)
+        try:
+            print(f"Joining existing room: {room_url}")
+            room: DailyRoomObject = await daily_helpers["rest"].get_room_from_url(room_url)
+        except Exception:
+            raise HTTPException(
+                status_code=500, detail=f"Room not found: {room_url}")
+
+    # Give the agent a token to join the session
+    token = await daily_helpers["rest"].get_token(room.url, MAX_SESSION_TIME)
+
+    if not room or not token:
+        raise HTTPException(
+            status_code=500, detail="Failed to get room or token token")
+
+    # Spawn a new agent, and join the user session
+    # Note: this is mostly for demonstration purposes (refer to 'deployment' in docs)
+    bot_proc = f"{sys.executable} -m bot_daily -u {room.url} -t {token} -i {callId} -d {callDomain} -p {phone_number}"
+
+    try:
+        subprocess.Popen(
+            [bot_proc],
+            bufsize=1,
+            cwd=os.path.dirname(os.path.abspath(__file__)),
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Failed to start subprocess: {e}")
+
+    return room
+
+
+@app.post("/outbound_call")
+async def outbound_call(request: Request) -> JSONResponse:
+    # Get the phone number from the request body
+    data = await request.json()
+    phone_number = data.get("phone_number")
+
+    if not phone_number:
+        raise HTTPException(status_code=400, detail="Phone number is required")
+
+    # create callId
+    callId = str(uuid.uuid4())
```

Comment:
callId and callDomain are not needed in the case of dialout.

It is only needed in the case of pinless dialin, because the `callId` and `callDomain` are SIP call legs that are not yet associated with a Daily Room. When the Daily room becomes ready to ingest the call, the two values are used to bridge the call